### PR TITLE
Add nvdimm lifecycle test case

### PR DESCRIPTION
**Pull request by jinqi7**
Signed-off-by: qijing <jinqi@redhat.com>

### CODE REVIEW

**Comment by jinqi7** [comment]
avocado run --vt-type libvirt nvdimm.back_file.label.check_life_cycle
JOB ID : 66c4597fcb1447b3017ffb5110019114f2709f1c
JOB LOG : /root/avocado/job-results/job-2020-06-01T06.02-66c4597/job.log
(1/1) type_specific.io-github-autotest-libvirt.nvdimm.back_file.label.check_life_cycle: PASS (78.12 s)
RESULTS : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME : 79.63 s

**Comment by jinqi7** [comment]
@chloerh,  can you please help to review it?

File: libvirt/tests/src/memory/nvdimm.py [hunk 1]
```diff
@@ -96,6 +96,17 @@ def create_nvdimm_xml(**mem_param):
...
         return mem_xml.copy()
 
+    def check_nvdimm_file(file_name):
+        """
+        check if the file exists in nvdimm memory device
+
+        :param file_name: the file name in nvdimm device
+        """
+        vm_session = vm.wait_for_login()
+        #vm_session.cmd('mount -o dax /dev/pmem0 /mnt/')
```
Comment (chunfuwen) [inline libvirt/tests/src/memory/nvdimm.py:106]: remote this comment-out code line

Comment (jinqi7) [reply libvirt/tests/src/memory/nvdimm.py]: Removed

File: libvirt/tests/src/memory/nvdimm.py [hunk 2]
```diff
@@ -96,6 +96,16 @@ def create_nvdimm_xml(**mem_param):
...
         return mem_xml.copy()
 
+    def check_nvdimm_file(file_name):
+        """
+        check if the file exists in nvdimm memory device
+
+        :param file_name: the file name in nvdimm device
+        """
+        vm_session = vm.wait_for_login()
+        if test_str not in vm_session.cmd('cat /mnt/%s ' % file_name):
```
Comment (sathnaga) [inline libvirt/tests/src/memory/nvdimm.py:106]: suggestion: we can pass mount path to the function and define mount path globally to the test rather using `/mnt` in multiple places... we can even change it different path later if needed...

can be addressed in separate PR aswell, I leave that to you. :-)

Comment (jinqi7) [reply libvirt/tests/src/memory/nvdimm.py:106]: Agree. It's better to pass the directory params in to the script.  But as you said, it's better to have  another PR to change it all :-)

File: libvirt/tests/src/memory/nvdimm.py [hunk 3]
```diff
@@ -96,6 +96,16 @@ def create_nvdimm_xml(**mem_param):
...
+            test.fail('"%s" should be in output' % test_str)
```
Comment (kylazhang) [inline libvirt/tests/src/memory/nvdimm.py:107]: Better more clear, should be in what output?

File: libvirt/tests/src/memory/nvdimm.py [hunk 4]
```diff
@@ -234,6 +244,26 @@ def create_nvdimm_xml(**mem_param):
...
+                libvirt.check_exit_status(result, expect_error=False)
+                check_nvdimm_file('foo-label')
+
+                result = virsh.snapshot_create_as(vm_name, vm_s1, ignore_status=True, debug=True)
+                libvirt.check_exit_status(result, expect_error=False)
+                result = virsh.snapshot_revert(vm_name, vm_s1, ignore_status=True, debug=True)
+                libvirt.check_exit_status(result, expect_error=False)
+                result = virsh.snapshot_delete(vm_name, vm_s1, ignore_status=True, debug=True)
+                libvirt.check_exit_status(result, expect_error=False)
+
```
Comment (kylazhang) [inline libvirt/tests/src/memory/nvdimm.py:266]: If all expect_error=False, why not just ignore_status=False and no need the check_exit_status?

Comment (jinqi7) [reply libvirt/tests/src/memory/nvdimm.py:260]: Changed.

File: libvirt/tests/src/memory/nvdimm.py (hunk 3 above)
Comment (jinqi7) [reply libvirt/tests/src/memory/nvdimm.py:107]: Changed it to the "output of cat cmd".

File: libvirt/tests/cfg/memory/nvdimm.cfg [hunk 5]
```diff
@@ -30,6 +30,10 @@
                     setvm_vcpu = 4
                     setvm_placement = static
                     qemu_checks = -numa node,nodeid=1,cpus=2-3,mem=512`mem-path=${nvdimm_file},share=yes,size=536870912`label-size=262144
+                    variants:
+                        - no_other_check:
```
Comment (chloerh) [inline libvirt/tests/cfg/memory/nvdimm.cfg:34]: Let's replace 'no_other_check' with 'default', shall we?

Comment (chloerh) [reply libvirt/tests/cfg/memory/nvdimm.cfg:34]: I'll make the change after merged

**Pull request merged**